test/Interpreter/generic_casts.swift

```diff
@@ -157,27 +158,104 @@ func genericAnyIs<T>(type: Any.Type, to: T.Type, expected: Bool) -> Bool {
     return expected
   }
 }
+// `as?` checks
```

Comment:
In theory `is`, `as?`, and `as!` all do variations of the same thing, but in practice there are different code paths for each one.  Indeed, one of the bugs I'm fixing in this PR is a discrepancy between `as?` and `as!` handling.  That's why I duplicated the existing `*AnyIs*` checks as `*AnyAsConditional*` and `*AnyAsUnconditional*`.